aws/service.ts

```diff
@@ -699,44 +660,43 @@ export class Service extends pulumi.ComponentResource implements cloud.Service {
         // Get the network to create the Service within.
         const network = getOrCreateNetwork();
 
-        // Create load balancer listeners/targets for each exposed port.
+        // AWS only supports a single load balancer per Service, as per:
+        // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-load-balancing.html
+        //
+        // So we find the single container that wants to expose a port (and throw if we see more
+        // than one).  We then create a load balancer that maps accordingly.
+
+        // We only use an array here to properly serialize into the aws.ecs.Service we're creating
+        // below.
         const loadBalancers = [];
 
-        let firstContainerName: string | undefined;
-        let firstContainerPort: number | undefined;
+        const singleContainerWithPortOpt = singleContainerWithExposedPort(args.containers!);
+        let exposedPortOpt: ExposedPort | undefined;
 
-        for (const containerName of Object.keys(containers)) {
-            const container = containers[containerName];
-            if (firstContainerName === undefined) {
-                firstContainerName = containerName;
-                if (container.ports && container.ports.length > 0) {
-                    firstContainerPort = container.ports[0].port;
-                }
-            }
+        if (singleContainerWithPortOpt) {
+            const portMapping = singleContainerWithPortOpt.container.ports![0];
 
-            ports[containerName] = {};
-            if (container.ports) {
-                for (const portMapping of container.ports) {
-                    if (loadBalancers.length > 0) {
-                        throw new Error("Only one port can currently be exposed per Service.");
-                    }
-                    const info = createLoadBalancer(this, cluster, name, containerName, portMapping, network);
-                    ports[containerName][portMapping.port] = {
-                        host: info.loadBalancer,
-                        hostPort: portMapping.port,
-                        hostProtocol: info.protocol,
-                    };
```

Comment:
@lukehoban This part is definitely the most suspect for me.  We were setting up this mapping from "container -> port -> hostPost", except 'port' and 'hostPort' are always the same here.

Then, later on we read these values out here: https://github.com/pulumi/pulumi-cloud/pull/661/files#diff-411860beb3c2bdcbd01894df3b3bd69aL239

we enumerate these and place 'port' into 'port' and 'hostPort' into 'hostport'.  So these values are all the same.  

This seem very fishy to me.  As you can see in other parts of the code like 3 lines below, we often use '.targetPort' if provided.  Is this all expected for you?  Or do you think we have a bug here?